File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = {};
        
        String[] nums = s.replaceAll("[{}]", "").split(",");
        Map<Integer, Integer> m = new HashMap<>();
        
        for(String n : nums){
            int tmp = Integer.parseInt(n.trim());
            m.put(tmp, m.getOrDefault(tmp, 0) + 1);
        }
        
        List<Integer> list = new ArrayList<>(m.keySet());
        list.sort((a, b) -> m.get(b) - m.get(a));
        
        return answer = list.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    l = list(s.strip('{{').strip('}}').split('},{'))
    l.sort(key=len)
    for i in range(len(l)):
        c = list(map(int, l[i].split(',')))
        for j in range(len(c)):
            if c[j] not in answer:
                answer.append(c[j])
    
    return answer